lib/tpm2_identity_util.c

```diff
@@ -481,16 +489,20 @@ bool tpm2_identity_create_name(TPM2B_PUBLIC *public, TPM2B_NAME *pubname) {
     }
 
     // Step 3 - Hash the data into name just past the alg type.
-    digester d = tpm2_openssl_halg_to_digester(name_alg);
-    if (!d) {
+    const EVP_MD *md = tpm2_openssl_md_from_tpmhalg(name_alg);
+    if (!md) {
+        LOG_ERR("Algorithm not supported: %x", name_alg);
         return false;
     }
 
-    d((const unsigned char *) &marshaled_tpmt, tpmt_marshalled_size,
-            pubname->name + hash_offset);
+    unsigned int hash_size;
+    if (!EVP_Digest(&marshaled_tpmt, tpmt_marshalled_size,
```

Comment:
We don't hide function returns in conditionals. This and elsewhere...

Reply:
OK, I have fixed that.

No big deal here, but openssl heavily uses "streams of commands" also elsewhere, relying on lazy evaluation. For example, see the code at the end of this page: https://www.openssl.org/docs/manmaster/man3/EVP_PKEY_fromdata.html.
It is of course possible to rewrite it with explicit `rc = `, but the beauty of a concise code will be lost.